ni_measurementlink_service/session_management/_reservation.py

```diff
@@ -535,21 +496,19 @@ def _initialize_multiplexer_sessions_core(
         self,
         session_constructor: Callable[[MultiplexerSessionInformation], TMultiplexerSession],
         multiplexer_type_id: Optional[str],
-        multiplexer_resource_names: Union[str, Iterable[str], None],
         closing_function: Optional[
             Callable[[TMultiplexerSession], ContextManager[TMultiplexerSession]]
         ] = None,
     ) -> Generator[Sequence[TypedMultiplexerSessionInformation[TMultiplexerSession]], None, None]:
         _check_optional_str_param("multiplexer_type_id", multiplexer_type_id)
         multiplexer_session_infos = self._validate_and_get_matching_multiplexer_session_infos(
-            _to_iterable(multiplexer_resource_names, self._multiplexer_resource_names),
             _to_iterable(multiplexer_type_id, self._multiplexer_type_ids),
         )
-
-        # If the user specified both the matching criteria, validate that matches were found.
-        self._validate_matched_multiplexer_sessions_with_resource_names(
-            multiplexer_session_infos, _to_iterable(multiplexer_resource_names)
-        )
+        if len(multiplexer_session_infos) == 0:
+            raise ValueError(
+                f"No multiplexer sessions matched type ID '{multiplexer_type_id}'. "
+                "Expected single session, got 0 sessions."
```

Comment:
`Expected single or multiple sessions, got 0 sessions.`

Reply:
This check is redundant as this [check](https://github.com/ni/measurementlink-python/blob/7aaba2b3744655be0eded07cc5b29d1f03d285ab/ni_measurementlink_service/session_management/_reservation.py#L388-L390) will validate that case. Hence, I've removed it now. Thanks for pointing it out.

A unit [test](https://github.com/ni/measurementlink-python/blob/7aaba2b3744655be0eded07cc5b29d1f03d285ab/tests/unit/test_reservation_multiplexer.py#L171) is in place showcasing that behavior.